os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -275,65 +275,74 @@ static void up_dumpstate(void)
 
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
+
 	lldbg("sp:     %08x\n", sp);
 
-#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
-	/* Get the limits on the nested irq stack */
-	nestirqstkbase = (uint32_t)&g_nestedirqstkbase;
-	nestirqstksize = (CONFIG_ARCH_NESTED_IRQ_STACK_SIZE & ~3);
+	/* Show the interrupt number, handler and stack information
+	 * if we are in an interrupt handler
+	 */
 
-	lldbg("Nested IRQ stack:\n");
-	lldbg("  base: %08x\n", nestirqstkbase);
-	lldbg("  size: %08x\n", nestirqstksize);
+	if (current_regs) {
+#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
+		lldbg("Code asserted in nested IRQ state!\n");
+#else
+		lldbg("Code asserted in IRQ state!\n");
+#endif
+		lldbg("IRQ num: %d \n", getipsr());
+		lldbg("IRQ handler: %08x \n", g_irqvector[getipsr()].handler);
+#ifdef CONFIG_DEBUG_IRQ_INFO
+		lldbg("IRQ handler: %s \n", g_irqvector[getipsr()].irq_name);
+#endif
+#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
+		/* Get the limits on the nested irq stack */
+		nestirqstkbase = (uint32_t)&g_nestedirqstkbase;
+		nestirqstksize = (CONFIG_ARCH_NESTED_IRQ_STACK_SIZE & ~3);
+	
+		lldbg("Nested IRQ stack:\n");
+		lldbg("  base: %08x\n", nestirqstkbase);
+		lldbg("  size: %08x\n", nestirqstksize);
 #ifdef CONFIG_STACK_COLORATION
-	lldbg("  used: %08x\n", up_check_nestirqstack());
+		lldbg("  used: %08x\n", up_check_nestirqstack());
```

Comment:
How about printing this info only if the stack pointer is within the limits? I am not sure of what purpose this serves when the sp is in a different stack.